app/views/users/_following_modal.html.erb

```diff
@@ -18,7 +18,7 @@
               <% end %>
             </ul>
           <% else %>
-            <% if @user.id == current_user.id %> 
+            <% if current_user and @user.id == current_user.id %>
```

Comment:
@jcoyne unfortunately no.  If the current_user is nil and the @user is nil current_user == @user evaluates to true which gives us "You..." instead of "This user..."